### PR TITLE
Dockerfile: Add Package - Install openmpi for ROCm images

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Install openmpi-4.0.0 for ROCm images.

### CODE REVIEW

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile [hunk 1]
```diff
@@ -41,6 +42,18 @@ RUN wget -qO - http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | APT_KEY_DON
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* /tmp/*
 
+# Install Open MPI
+ENV OPENMPI_VERSION=4.0.0
```
Comment (abuccts) [inline dockerfile/rocm4.2-pytorch1.7.0.dockerfile:46]: use the same version 4.0.5 as cuda image?

**Review by abuccts — APPROVED** [review]

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile (hunk 1 above)
Comment (guoshzhao) [reply dockerfile/rocm4.2-pytorch1.7.0.dockerfile]: Done.

File: dockerfile/cuda11.1.1.dockerfile [hunk 2]
```diff
@@ -2,6 +2,7 @@ FROM nvcr.io/nvidia/pytorch:20.12-py3
 
 # OS:
 #   - Ubuntu: 20.04
+#   - OpenMPI: 4.0.5
```
Comment (cp5555) [inline dockerfile/cuda11.1.1.dockerfile:5]: Where is the OpenMPI installation?

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile [hunk 3]
```diff
@@ -41,6 +42,18 @@ RUN wget -qO - http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | APT_KEY_DON
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* /tmp/*
 
+# Install Open MPI
```
Comment (cp5555) [inline dockerfile/rocm4.2-pytorch1.7.0.dockerfile:45]: # Intsall OpenMPI

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile [hunk 4]
```diff
@@ -41,6 +42,18 @@ RUN wget -qO - http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | APT_KEY_DON
...
+ENV OPENMPI_VERSION=4.0.5
+RUN cd /tmp && \
+    wget -q https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-${OPENMPI_VERSION}.tar.gz && \
+    tar xzf openmpi-${OPENMPI_VERSION}.tar.gz && \
+    cd openmpi-${OPENMPI_VERSION} && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig && \
+    rm -rf openmpi-${OPENMPI_VERSION}*
```
Comment (cp5555) [inline dockerfile/rocm4.2-pytorch1.7.0.dockerfile:55]: Do we need to do `cd ..` first?

File: dockerfile/rocm4.0-pytorch1.7.0.dockerfile [hunk 5]
```diff
@@ -41,6 +42,18 @@ RUN wget -qO - http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | APT_KEY_DON
...
+ENV OPENMPI_VERSION=4.0.5
+RUN cd /tmp && \
+    wget -q https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-${OPENMPI_VERSION}.tar.gz && \
+    tar xzf openmpi-${OPENMPI_VERSION}.tar.gz && \
+    cd openmpi-${OPENMPI_VERSION} && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig && \
+    rm -rf openmpi-${OPENMPI_VERSION}*
```
Comment (cp5555) [inline dockerfile/rocm4.0-pytorch1.7.0.dockerfile:55]: Do we need to do `cd ..` first?

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile (hunk 3 above)
Comment (guoshzhao) [reply dockerfile/rocm4.2-pytorch1.7.0.dockerfile:45]: Done.

File: dockerfile/cuda11.1.1.dockerfile (hunk 2 above)
Comment (guoshzhao) [reply dockerfile/cuda11.1.1.dockerfile:5]: It will be installed into system path: `/usr/local/bin/`

Comment (cp5555) [reply dockerfile/cuda11.1.1.dockerfile:5]: I mean there is no code for OpenMPI installation.

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile (hunk 4 above)
Comment (guoshzhao) [reply dockerfile/rocm4.2-pytorch1.7.0.dockerfile:55]: Good catch

File: dockerfile/rocm4.0-pytorch1.7.0.dockerfile (hunk 5 above)
Comment (guoshzhao) [reply dockerfile/rocm4.0-pytorch1.7.0.dockerfile:55]: change to `rm -rf /tmp/openmpi-${OPENMPI_VERSION}*`

File: dockerfile/rocm4.2-pytorch1.7.0.dockerfile (hunk 4 above)
Comment (guoshzhao) [reply dockerfile/rocm4.2-pytorch1.7.0.dockerfile:55]: change to `rm -rf /tmp/openmpi-${OPENMPI_VERSION}*`

File: dockerfile/cuda11.1.1.dockerfile (hunk 2 above)
Comment (guoshzhao) [reply dockerfile/cuda11.1.1.dockerfile:5]: For nvidia image, it is installed in base image. Just list the version in comment.

**Review by cp5555 — APPROVED** [review]

**Review by yukirora — APPROVED** [review]

**Pull request merged**